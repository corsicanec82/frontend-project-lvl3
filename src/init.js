export default () => {
  const form = document.querySelector('div.jumbotron form');
  const p = document.createElement('p');
  p.textContent = 'description for jumbotron';
  form.before(p);
};
