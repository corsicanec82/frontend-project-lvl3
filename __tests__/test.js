import { promises as fs } from 'fs';
import path from 'path';
import { html } from 'js-beautify';
import init from '../src/init';

const createDOM = (htmlContent) => {
  const dom = document.implementation.createHTMLDocument();
  dom.open();
  dom.write(htmlContent);
  dom.close();
  const htmlElement = dom.querySelector('html');
  document.documentElement.remove();
  document.append(htmlElement);
};

const getTree = () => html(document.documentElement.outerHTML);

describe('init testing', () => {
  beforeAll(() => {
    const pathToHtml = path.resolve(__dirname, '__fixtures__/index.html');
    return fs.readFile(pathToHtml, 'utf8').then(createDOM);
  });

  it('init', () => {
    init();
    expect(getTree()).toMatchSnapshot();
  });
});
