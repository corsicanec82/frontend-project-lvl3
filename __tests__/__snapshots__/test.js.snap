// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init testing init 1`] = `
"<html lang=\\"en\\">

<head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\">
    <title>RSS Reader</title>
</head>

<body>
    <div class=\\"container\\">
        <div class=\\"jumbotron\\">
            <h1>Form to add RSS feed</h1>
            <br>
            <p>description for jumbotron</p>
            <form>
                <div class=\\"form-group row\\">
                    <div class=\\"col\\">
                        <label class=\\"sr-only\\" for=\\"inputRSS\\">label</label>
                        <input type=\\"text\\" class=\\"form-control\\" id=\\"inputRSS\\" placeholder=\\"Enter data to add rss feed\\">
                    </div>
                    <div class=\\"col\\">
                        <button type=\\"submit\\" class=\\"btn btn-primary\\">Add feed</button>
                    </div>
                </div>
            </form>
        </div>
    </div>


</body>

</html>"
`;
